packages/andromeda-non-fungible-tokens/src/marketplace.rs

```diff
@@ -0,0 +1,98 @@
+use common::ado_base::{modules::Module, AndromedaMsg, AndromedaQuery};
+use cosmwasm_std::{Addr, Uint128};
+use cw721::Cw721ReceiveMsg;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct InstantiateMsg {
+    pub modules: Option<Vec<Module>>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    AndrReceive(AndromedaMsg),
+    ReceiveNft(Cw721ReceiveMsg),
+    /// Transfers NFT to buyer and sends funds to seller
+    Buy {
+        token_id: String,
+        token_address: String,
+    },
+    /// Updates the sale's price, demomination, and whitelist
+    UpdateSale {
+        token_id: String,
+        token_address: String,
+        price: Uint128,
+        coin_denom: String,
+        whitelist: Option<Vec<Addr>>,
```

Comment:
Do we need this if the contract has modules? Or do we maybe want to consider using our addresslist module instead?

Reply:
Yes actually using the address-list module could replace the whitelist option